tests/test_recursive_number_partitioning_v2.py

```diff
@@ -0,0 +1,82 @@
+    assert _compare_partitions(partition(algorithm=rnp, numbins=3, items={'a': 1, 'b': 7, 'c': 9, 'd': 2}), [['c'], ['b'], ['d', 'a']], valueof={'a': 1, 'b': 7, 'c': 9, 'd': 2}.__getitem__)
+    assert _compare_partitions(partition(algorithm=rnp, numbins=3, items=['a', 'b', 'c', 'd'], valueof=lambda c: ord(c) ** 2), [['d'], ['c'], ['b', 'a']], valueof=lambda c: ord(c) ** 2)
+
+
+def test_4way_partitioning():
+    assert _compare_partitions(partition(algorithm=rnp, numbins=4, items=[1, 7, 9, 2, 6, 7]), [[9], [7, 2], [7, 1], [6]])
+    assert _compare_partitions(partition(algorithm=rnp, numbins=4, items=[2, 2]), [[2], [2], [], []])
+    assert _compare_partitions(partition(algorithm=rnp, numbins=4, items=[2, 2, 2]), [[2], [2], [2], []])
+    assert _compare_partitions(partition(algorithm=rnp, numbins=4, items=[]), [[], [], [], []])
+    assert _compare_partitions(partition(algorithm=rnp, numbins=4, items=list(range(1, 11))), [[10, 4], [9, 5], [8, 6], [7, 3, 2, 1]])
+    assert _compare_partitions(partition(algorithm=rnp, numbins=4, items={'a': 1, 'b': 7, 'c': 9, 'd': 2, 'e': 6, 'f':7}), [['c'], ['b', 'd'], ['f', 'a'], ['e']], valueof={'a': 1, 'b': 7, 'c': 9, 'd': 2, 'e': 6, 'f':7}.__getitem__)
+    assert _compare_partitions(partition(algorithm=rnp, numbins=4, items=['a', 'b', 'c', 'd', 'e'], valueof=lambda c: ord(c) ** 2), [['e'], ['d'], ['c'], ['b', 'a']], valueof=lambda c: ord(c) ** 2)
+
+
+def test_1way_partitioning():
+    assert _compare_partitions(partition(algorithm=rnp, numbins=1, items=[1, 2, 3]), [[3, 2, 1]])
+    assert _compare_partitions(partition(algorithm=rnp, numbins=1, items={c: ord(c) - 97 for c in string.ascii_lowercase}), [list(string.ascii_lowercase)], valueof=lambda c: ord(c) - 97)
+    assert partition(algorithm=rnp, numbins=1, items=[100]) == [[100]]
+    assert partition(algorithm=rnp, numbins=1, items=[]) == [[]]
+
+
+def test_zero():
+    assert _compare_partitions(partition(algorithm=rnp, numbins=2, items=[0, 0, 0, 0]), [[0, 0, 0, 0], []])
+    assert _compare_partitions(partition(algorithm=rnp, numbins=3, items=[0, 0, 0, 0]), [[0, 0, 0, 0], [], []])
+    assert _compare_partitions(partition(algorithm=rnp, numbins=4, items=[0, 0, 0, 0]), [[0, 0, 0, 0], [], [], []])
+
+    assert _compare_partitions(partition(algorithm=rnp, numbins=2, items=[0, 0, 0, 0, 6, 0, 0, 3, 0]), [[6, 0, 0, 0, 0, 0, 0, 0], [3]])
+    assert _compare_partitions(partition(algorithm=rnp, numbins=3, items=[0, 0, 0, 0, 6, 0, 0, 3, 0]), [[6, 0, 0, 0, 0, 0, 0, 0], [3], []])
+
+    assert _compare_partitions(partition(algorithm=rnp, numbins=3, items=['a', 'b', 'c', 'd'], valueof=lambda c: ord(c)*(ord(c)%2)), [['c', 'b', 'd'], ['a'], []], valueof=lambda c: ord(c)*(ord(c)%2))
+
+
+def test_on_random_inputs():
+    # compare to integer programming (optimal solution)
+    for numbins in [2, 3, 4]:
+        items = [random.randint(1, 256) for _ in range(5)]
```

Comment:
Test on slightly larger instances. An instance with 10 items should still run quite fast.